specification.markdown

```diff
@@ -155,11 +155,39 @@ Class D    | 4      | 0x101FFFFF - 0x1FFFFFFE |   268,338,304
+## Unknown Data Size
+
+An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)).
+
+The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first:
+
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a Parent Element of the Unknown-Sized Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is not a Descendant Element of the Unknown-Sized Element but share a common direct parent. For example a Top-Level Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a root element.
+- The end of the EBML Document, either when reaching the end of the file or because a new EBML Header started.
+
+Consider an Unknown-Size Element which EBML path is `\root\level1\level2\elt`. Here are some possible and impossible way to end this element:
+
+EBML Path of next element          | Status
+-----------------------------------|------------------------------
+`\root\level1\level2`              | Ends the unknown-size element, as it is a new Parent element
+`\root\level1`                     | Ends the unknown-size element, as it is a new Parent element
+`\root`                            | Ends the unknown-size element, as it is a new root element
+`\root2`                           | Ends the unknown-size element, as it is a new root element
+`\root\level1\level2\other`        | Ends the unknown-size element, as they share the same parent
+`\root\level1\level2\elt\inside`   | Doesn't end the unknown-size element, it's a child of `elt`
+`\root\level1\level2\other\part`   | Isn't valid right after `elt`. `other` should be found first
+`\root\level1\level2\elt\<global>` | Global Element is valid, it's a child of `elt`
+`\root\level1\level2\<global>`     | Global Element is invalid, it can only be a child of `elt` while parsing `elt`
```

Comment:
I'm confused by this wording. The global element here seems to be placed in a valid way, but I think you're trying to see that a global can't be used to consider where the end of `elt` is. Actually in these two cases, it seems impossible to distinguish the parent of the global element.

Reply:
Sorry, @robUx4, still lost on how to interpret this. I think if you are reading `elt` (unknown size) and a Global Element is found, it is unknown if the Global Element is a Child Element or a new Element that follows the end of the elt element. Is there a way to clarify?

Reply:
so maybe we say "Global Element is valid, but it is indecipherable if the Global Element is a Child Element of `elt` or an Element that follows the end of the `elt` Element."

Reply:
Yes, this case is not "invalid" as in "it cannot happen here", it just can't be interpreted this way because if you find a `Global Element` it's interpreted as the example above and no other way.